lib/RESTClient.js

```diff
@@ -351,9 +352,12 @@ class RESTClient {
                 res.on('data', data => {
                     ret.push(data);
                     retLen += data.length;
-                }).on('error', callback)
+                }).on('error', error => {
+                    log.error('error receiving response from metadata', { error });
```

Comment:
we should toString the error or add the stacktrace, by default the stringify won't pick the error message. You might only have the code if there is one such a `{code: 'ECONNRESET'}`

```suggestion
                    log.error('error receiving response from metadata', { error, msg: error.toString() });
```

Reply:
done (squashed in the main commit)